File: iosApp/iosApp/Services/ImprovementProposalsServiceAssembler.swift
// Created by web3d4v on 27/07/2022.
// Copyright (c) 2022 Sons Of Crypto.
// SPDX-License-Identifier: MIT

import Foundation
import web3walletcore

final class ImprovementProposalsServiceAssembler: AssemblerComponent {

    func register(to registry: AssemblerRegistry) {
        registry.register(scope: .instance) { resolver -> ImprovementProposalsService in
            DefaultImprovementProposalsService(
                store: KeyValueStore(name: "\(ImprovementProposalsService.self)")
            )
        }
    }
}

File: androidApp/src/main/java/com/sonsofcrypto/web3wallet/android/services/ImprovementProposalsServiceAssembler.kt
package com.sonsofcrypto.web3wallet.android.services

import com.sonsofcrypto.web3lib.keyValueStore.KeyValueStore
import com.sonsofcrypto.web3wallet.android.common.AssemblerComponent
import com.sonsofcrypto.web3wallet.android.common.AssemblerRegistry
import com.sonsofcrypto.web3wallet.android.common.AssemblerRegistryScope
import com.sonsofcrypto.web3walletcore.services.improvementProposals.DefaultImprovementProposalsService
import com.sonsofcrypto.web3walletcore.services.improvementProposals.ImprovementProposalsService
import smartadapter.internal.extension.name

class ImprovementProposalsServiceAssembler: AssemblerComponent {

    override fun register(to: AssemblerRegistry) {

        to.register(ImprovementProposalsService::class.name, AssemblerRegistryScope.SINGLETON) {

            DefaultImprovementProposalsService(
                it.resolve("${KeyValueStore::class.name}.ImprovementProposalsService"),
            )
        }
    }
}